### PR TITLE
feat: SearchStateController class

### PR DESCRIPTION
**Pull request by mohdashraf010897**
**PR Type** `Feature` ⭐ 

**Description** The PR adds `SearchStateController` class to the `searchbase`-dart library. Additionally, `StateProvider` now utilizes the `SearchStateController` class to manage internal state.

[📔  Notion](https://www.notion.so/appbase/StateProvider-Usage-without-onChange-and-build-methods-d4fe55249b77439ab7377457b2230762)

https://www.loom.com/share/141ed8488a194277954454d29084bcaa
https://www.loom.com/share/077a340b0d77472fb33c1217215e15da
--- 
**Docs PR** https://github.com/appbaseio/Docs/pull/316

### CODE REVIEW

File: searchbase/README.md [hunk 1]
```diff
@@ -248,3 +248,214 @@ class DefaultUriPolicy implements UriPolicy {
   }
 }
 ```
+
+### Facet example extended to use SearchStateController to handle state changes side-effects
+
+This example is a version of the previous example which uses `SearchStateController` class to watch for state changes and perform relevant side-effects.
```
Comment (bietkul) [inline searchbase/README.md]: > previous example

Can you link the example?

File: searchbase/lib/src/searchStateController.dart [hunk 2]
```diff
@@ -0,0 +1,336 @@
+import 'package:searchbase/searchbase.dart';
+import 'types.dart';
+import 'searchcontroller.dart';
+import 'constants.dart';
+
+/// It stores the state of the SearchController at any time.
+class SearchControllerState {
+  final Aggregations? aggregationData;
```
Comment (bietkul) [inline searchbase/lib/src/searchStateController.dart:9]: Docs are missing for all the properties.

**Review by bietkul — CHANGES_REQUESTED** [review]

File: searchbase/README.md (hunk 1 above)
Comment (mohdashraf010897) [reply searchbase/README.md]: Fixed here 👉  https://github.com/appbaseio/flutter-searchbox/pull/21/commits/1bc12704ca626f05c9bceb0804e5dff385bae78c

File: searchbase/lib/src/searchStateController.dart (hunk 2 above)
Comment (mohdashraf010897) [reply searchbase/lib/src/searchStateController.dart:9]: Fixed here 👉  https://github.com/appbaseio/flutter-searchbox/pull/21/commits/e43591b8d06971646f10850237673cc68e5b4e9c

**Review by bietkul — APPROVED** [review]

**Pull request merged**